### PR TITLE
ENT-3137 | Created management command for sending offer usage alerts.

### PR DESCRIPTION
**Pull request by hasnain-naveed**

### CODE REVIEW

File: ecommerce/enterprise/management/commands/send_enterprise_offer_limit_emails.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        for enterprise_offer in enterprise_offers:
+            if self.is_eligible_for_alert(enterprise_offer):
+                send_enterprise_offer_count += 1
+                email_body, email_subject = self.get_email_content(enterprise_offer)
+                OfferUsageEmail.create_record(enterprise_offer, meta_data={
+                    'email_body': email_body,
+                    'email_subject': email_subject,
+                    'email_addresses': enterprise_offer.emails_for_usage_alert
+                })
+                send_offer_assignment_email.delay(enterprise_offer.emails_for_usage_alert, email_subject, email_body)
```
Comment (hasnain-naveed) [inline ecommerce/enterprise/management/commands/send_enterprise_offer_limit_emails.py]: I have to change this function with my function. But right now, ecommerce-worker hasn't deployed yet.

File: ecommerce/enterprise/management/commands/send_enterprise_offer_limit_emails.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+OfferUsageEmail = get_model('offer', 'OfferUsageEmail')
+OrderDiscount = get_model('order', 'OrderDiscount')
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+EMAIL_BODY = """
+    You have used {percentage_usage}% of the {offer_type} Limit associated with the entitlement offer called 
+    "{offer_name}"\n {offer_type}s Redeemed: {current_usage}\n {offer_type}s Limit: {total_limit}\n Please reach out to 
+    customersuccess@edx.org, or to your Account Manager or Customer Success representative, if you have any questions.
```
Comment (irfanuddinahmad) [inline ecommerce/enterprise/management/commands/send_enterprise_offer_limit_emails.py]: So many 'or' in this message!

File: ecommerce/enterprise/management/commands/send_enterprise_offer_limit_emails.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            is_eligible = True if diff_of_days >= 1 else False
+        elif enterprise_offer.usage_email_frequency == ConditionalOffer.WEEKLY:
+            is_eligible = True if diff_of_days >= 7 else False
+        else:
+            is_eligible = True if diff_of_days >= 30 else False
+        return is_eligible
+
+    @staticmethod
+    def get_enrollment_limits(offer):
+        return (offer.num_orders/offer.max_global_applications) * 100, offer.num_orders
```
Comment (irfanuddinahmad) [inline ecommerce/enterprise/management/commands/send_enterprise_offer_limit_emails.py]: do we always have a value for max_global_applications?

File: ecommerce/enterprise/management/commands/send_enterprise_offer_limit_emails.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            offer_type='Enrollment' if is_enrollment_limit_offer else 'Booking',
+            offer_name=offer.name,
+            current_usage=int(current_usage)
+        )
+        return email_body, EMAIL_SUBJECT
+
+    @staticmethod
+    def _get_enterprise_offers():
+        """
+        Return the enterprise offer which have enabled for email usage alert.
```
Comment (irfanuddinahmad) [inline ecommerce/enterprise/management/commands/send_enterprise_offer_limit_emails.py]: Return the enterprise offers which have opted for email usage alert.

**Review by moconnell1453 — APPROVED** [review]

File: ecommerce/enterprise/management/commands/send_enterprise_offer_limit_emails.py (hunk 3 above)
Comment (hasnain-naveed) [reply ecommerce/enterprise/management/commands/send_enterprise_offer_limit_emails.py]: This function will only be called for the offers which have max_global_applications value.

File: ecommerce/enterprise/management/commands/send_enterprise_offer_limit_emails.py (hunk 2 above)
Comment (hasnain-naveed) [reply ecommerce/enterprise/management/commands/send_enterprise_offer_limit_emails.py]: This is the message sent by Nathan and he has approved on the story.

**Review by irfanuddinahmad — APPROVED** [review]
OK when green!

**Pull request merged**